Update to latest sewing-kit

BPScott opened the pull request:
### WHY are these changes introduced?

Keeping up to date, includes new linting rules mostly, meaning when eslint-plugin-shopify jumps to use the new typescript parser there will be less effort to update it and get onto a newer typescript version.

### WHAT is this pull request doing?
Update a few dependencies that we want to keep in sync with sewing-kit

Use `sewing-kit optimize` to optimize svgs instead of maintaining our
own script which does the same thing.

---

An update to babel-plugin-shopify has changed the default value of "useBuiltIns". We now explicitly set "useBuildIns: 'entry'" in our babel.config.js to maintain the same behaviour as previously. Not doing this results in adding ~50kb extra polyfill stuff.

This doesn't affect polaris when used as part of sewing-kit because sk doesn't use the stand-alone build.

### To Review

This PR is split into multiple commits, one of which is all the automatic fixes from running `yarn run format` which contains the majority of the code changes (apart from the yarn.lock file). It might be easier to look at the commits one-by-one rather than scrolling through a load of automatic fixes.

### To Test

* Ensure tests / linting / type checks pass
* Ensure aria labels still appear as you'd expect in the components where I had to add translations

BPScott commented on .eslintrc:
```diff
@@ -56,20 +59,24 @@
       "files": ["**/*.test.{ts,tsx}"],
       "rules": {
+        "jest/no-truthy-falsy": "off",
+        "shopify/jsx-no-hardcoded-content": "off",
```
We want to check for hardcoded content to make sure we're using translated strings in our source, but we don't care about that in our tests and examples.

BPScott commented on .eslintrc:
```diff
@@ -28,13 +27,16 @@
     "shopify/jsx-no-complex-expressions": "off",
+    "shopify/jsx-prefer-fragment-wrappers": "off",
     "shopify/no-ancestor-directory-import": "error",
+    "shopify/react-prefer-private-members": "off",
```
Sometimes we do actually want pubic functions

AndrewMusgrave replied:
I wonder if we should enable this and add an ignore whenever public is used, which is only in `Banner` i think?

BPScott replied:
There's 36 violations currently, 1/3 of which are false positives warning about examples written in JS when you can't even use private members (bug raised: https://github.com/Shopify/eslint-plugin-shopify/issues/259), and many of the others seem related to legacy context stuff - subscribe/unsubscribe stuff. I don't know enough about the code they touch to know what should be private or not.

I'd totally be open to reevaluating this in the future, but I don't want to touch them now :)

BPScott commented on .eslintrc:
```diff
@@ -28,13 +27,16 @@
     "react/no-array-index-key": "off",
     "react/jsx-fragments": ["error", "element"],
     "shopify/jsx-no-complex-expressions": "off",
+    "shopify/jsx-prefer-fragment-wrappers": "off",
```
Sometimes we want empty wrapping divs

BPScott commented on .eslintrc:
```diff
@@ -28,13 +27,16 @@
         "allowBlockStart": false
       }
     ],
+    "babel/no-unused-expressions": "off",
```
This doesn't seem to work in the way that I'd expect

BPScott commented on .eslintrc:
```diff
@@ -56,20 +59,24 @@
     {
       "files": ["**/*.test.{ts,tsx}"],
       "rules": {
+        "jest/no-truthy-falsy": "off",
```
we use isTruthy() / isFalsy() A LOT and I don't want to go and refactor everything

AndrewMusgrave commented on src/components/Frame/components/Toast/tests/Toast.test.tsx:
```diff
@@ -113,18 +113,22 @@ describe('<Toast />', () => {
-      document.addEventListener = jest.fn((event, cb) => {
-        listenerMap[event] = cb;
-      });
+      jest
```
Should we clear this in an afterEach?

BPScott replied:
All these changes to use jest.spyOn are automatic fixes rather than manual ones. The end result is the same as the current behaviour. the symmetry of using jest.spyOn then mockRestore is nice but if i worked before it'll work after. I'd like to not get too bogged down in tidying up our tests as they work in the same way as before :)

I've updated the cases that you raised and found an extra one in Tabs.

AndrewMusgrave commented on src/components/Frame/components/Toast/tests/Toast.test.tsx:
```diff
@@ -113,18 +113,22 @@ describe('<Toast />', () => {
-      document.removeEventListener = jest.fn((event) => {
-        listenerMap[event] = noop;
-      });
+      jest
```
^

AndrewMusgrave commented on src/components/KeypressListener/tests/KeypressListener.test.tsx:
```diff
@@ -14,11 +14,11 @@ const listenerMap: HandlerMap = {};
 describe('<KeypressListener />', () => {
   beforeEach(() => {
-    document.addEventListener = jest.fn((event, cb) => {
+    jest.spyOn(document, 'addEventListener').mockImplementation((event, cb) => {
```
We'll want to change the afterEach to a clear rather than reassign the original value

AndrewMusgrave commented on src/components/Page/tests/Page.test.tsx:
```diff
@@ -170,7 +170,7 @@ describe('<Page />', () => {
 
       const {restore: restoreTitleBarCreateMock} = mockTitleBarCreate();
-      AppBridgeButton.create = jest.fn().mockReturnValue(buttonMock);
+      jest.spyOn(AppBridgeButton, 'create').mockReturnValue(buttonMock);
```
^

BPScott replied:
It looks like define restoration - see line 178

AndrewMusgrave commented on src/components/TextField/tests/TextField.test.tsx:
```diff
@@ -694,11 +694,13 @@ describe('<TextField />', () => {
-              documentEvent[eventType] = callback;
-            },
-          );
+          jest
```
^

BPScott replied:
Have changed the reassigning to use mockRestore().

AndrewMusgrave commented on src/components/TopBar/components/UserMenu/context/Modifier.tsx:
```diff
@@ -22,7 +22,7 @@ class Modifier extends React.Component<ComposedProps, {}> {
   }
 
-  state = {};
+  state: any = {};
```
We can probably 🔥 this and change line 14 to `never`

BPScott replied:
I've got no idea what this does. So I elected for not changing the code and just making typescript happy. It looks like this component still has state it's just defined in the passed setMobileUserMenuProps function.

Thinking a little deeper, matching this type with the second arg of `React.Component<ComposedProps, {}>` is a better plan. I've changed it to `state: {}`.

AndrewMusgrave commented:
I requested translations and since the translations have already been released feel free to mark as `async` when you want to 🚢

danrosenthal commented:
What's the status on this PR @BPScott?

BPScott commented:
I've rebased this atop master, fixed additional issues (autofixing jest's toEqual to toStrictEqual) and addressed PR feedback. I've also updated the babel config to retain the old useBuiltIns behaviour, which means the bundle size is unaffected (previously it grew by a substantial amount).

I've responded to your comments and this is now read for re-review @AndrewMusgrave

AndrewMusgrave commented on src/components/Frame/components/Toast/tests/Toast.test.tsx:
```diff
@@ -111,22 +111,40 @@ describe('<Toast />', () => {
 
-      listenerMap.keyup({keyCode: Key.Escape});
+      afterEach(() => {
+        (document.addEventListener as jest.Mock).mockRestore();
```
I'm ok with this, however an alternative could have been
```tsx
let removeEventListener: jest.SpyInstace;
...
removeEventListener = jest.spyOn(...)
...
removeEventListener.mockRestore
```
